Lib/dis.py

```diff
@@ -46,6 +47,10 @@ def dis(x=None, *, file=None, depth=None):
         x = x.__code__
     if hasattr(x, 'gi_code'):  # Generator
         x = x.gi_code
+    if hasattr(x, 'ag_code'):  # Async generator
```

Comment:
Agreed--thanks! I stuck with `if` to keep it consistent over the whole passage (since I didn't think through what was semantically correct for the rest of the comparison) but might as well decide now and clean it up.

Before changing line 46 to `elif`: is there any weird hybrid corner case where we would see both `__func__` and `__code__` and want `__code__` to override? I think that's impossible but I'm not 100% sure.

Also, if we change line 46 to `elif`, should we also change line 44 to `elif`? It's obviously unnecessary due to the `return`, but is it clearer? If not, I'd propose a space after line 43 to distinguish the `None` guard from the code-object indirection(s). Trying to tread lightly b/c that's technically outside of the patch.